### PR TITLE
remove unneeded AES conditionals now that we no longer support 0.9.8

### PR DESCRIPTION
**Pull request by reaperhulk**
refs #2982

### CODE REVIEW

File: src/_cffi_src/openssl/aes.py [hunk 1]
```diff
@@ -21,36 +21,24 @@
...
-   This should be when we drop RHEL/CentOS 5, which is on 0.9.8e. */
 int AES_wrap_key(AES_KEY *, const unsigned char *, unsigned char *,
                  const unsigned char *, unsigned int);
 int AES_unwrap_key(AES_KEY *, const unsigned char *, unsigned char *,
                    const unsigned char *, unsigned int);
+"""
 
-/* The ctr128_encrypt function is only useful in 0.9.8. You should use EVP for
-   this in 1.0.0+. It is defined in macros because the function signature
-   changed after 0.9.8 */
```
Comment (alex) [inline src/_cffi_src/openssl/aes.py]: Can this not be moved to FUNCTIONS?

Comment (reaperhulk) [inline src/_cffi_src/openssl/aes.py]: Possibly can. Let me look at the function definition for 1.0.0...

Comment (lvh) [inline src/_cffi_src/openssl/aes.py]: Does this mean the original comment was wrong, and you still need ctr128_encrypt for 1.0.0?

Comment (reaperhulk) [inline src/_cffi_src/openssl/aes.py]: Yep, you do. 1.0.1 is the first version with EVP AES CTR support.

**Comment by lvh** [comment]
Looks great to me 👍

**Comment by alex** [comment]
:+1: lgtm

**Pull request merged**